src/isStrictEqual.ts

```diff
@@ -0,0 +1,82 @@
+import { purry } from "./purry";
+
+/**
+ * Determines whether two values are *functionally identical* in all contexts.
+ * For primitive values (string, number), this is done by-value, and for objects
+ * it is done by-reference (i.e., they point to the same object in memory).
+ *
+ * Under the hood we use **both** the [`===` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality)
+ * and [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is). This means that `isEqual(NaN, NaN) === true` (whereas
+ * `NaN !== NaN`), and `isEqual(-0, 0) === true` (whereas
+ * `Object.is(-0, 0) === false`).
+ *
+ * The result would be narrowed to the second value so that the function can be
+ * used as a type guard.
+ *
+ * See:
+ * - `isDeepEqual` for a semantic comparison that allows comparing arrays and
+ * objects "by-value", and recurses for every item.
+ * - `isShallowEqual` if you need to compare arrays and objects "by-value" but
+ * don't want to recurse into their values.
+ *
+ * @param data - The first value to compare.
+ * @param other - The second value to compare.
+ * @signature
+ *    R.isEqual(data, other)
```

Comment:
Wasn't this meant to be R.isStrictEqual?

Reply:
It should be, the rename was last moment so I missed this... i'll send a fix.